### PR TITLE
`ChunkedPOST` in addition to `ChunkedGET`.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi @mzhurovich -- this is missing tests, but may well be good to go "as is", as I've confirmed it does the job (with `HTTP` examples from the [Docker API](https://docs.docker.com/engine/api/sdk/examples/)).

### CODE REVIEW

**Comment by dkorolev** [comment]
Upd: Per-line-parsing added, tests added, `Chunked{GET,POST}`-s extended to support dot notation -- good to review from my end. Thanks @mzhurovich!

File: bricks/strings/group_by_lines.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ private:
+  F f_;
+  std::string residual_;
+
+ public:
+  explicit GenericStatefulGroupByLines(F&& f) : f_(std::move(f)) {}
+  void Feed(const std::string& s) {
+    Feed(s.c_str());
+  }
+  void Feed(char const* s) {
```
Comment (mzhurovich) [inline bricks/strings/group_by_lines.h]: `const char*` for consistency?

File: blocks/http/types.h [hunk 2]
```diff
@@ -88,16 +88,97 @@ struct GET : HTTPRequestBase<GET> {
...
+    return static_cast<T&>(*this);
+  }
+  T& OnDone(std::function<void()> done_callback) {
+    this->done_callback_impl = done_callback;
+    group_by_lines_.clear();
+    return static_cast<T&>(*this);
+  }
+
+  // TODO(dkorolev): Move this to `Chunk`-s if performance becomes the bottleneck.
+  T& OnLine(std::function<void(const char*)> line_callback) {
```
Comment (mzhurovich) [inline blocks/http/types.h]: Hmm, not sure why the API for lines is `const char*`, while other functions are `const std::string&`

File: bricks/strings/test.cc [hunk 3]
```diff
@@ -991,3 +992,34 @@ TEST(UTF8StringLength, Smoke) {
   EXPECT_EQ(4u, current::strings::UTF8StringLength("test"));
   EXPECT_EQ(4u, current::strings::UTF8StringLength("тест"));
 }
+
+TEST(StatefulGroupByLines, SmokeTrivial) {
+  std::vector<std::string> lines;
+  StatefulGroupByLines splitter([&lines](const char* line) { lines.push_back(std::move(line)); });
```
Comment (mzhurovich) [inline bricks/strings/test.cc]: `.emplace_back(line)`?

File: bricks/strings/group_by_lines.h [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        residual_.clear();
+        ++s;
+      } else {
+        return;
+      }
+    }
+  }
+  ~GenericStatefulGroupByLines() {
+    if (!residual_.empty()) {
+      f_(residual_.c_str());
```
Comment (mzhurovich) [inline bricks/strings/group_by_lines.h:64]: Maybe add a comment explaining the behavior when there's no newline at the end of the string?

**Review by mzhurovich — COMMENTED** [review]
Thanks for adding this. Left a few minor comments/questions

File: bricks/strings/group_by_lines.h (hunk 1 above)
Comment (dkorolev) [reply bricks/strings/group_by_lines.h]: :+1: c5ec7c52

File: blocks/http/types.h (hunk 2 above)
Comment (dkorolev) [reply blocks/http/types.h]: Right, let's keep them `const std::string&`-s to keep things simple.

File: bricks/strings/group_by_lines.h (hunk 4 above)
Comment (dkorolev) [reply bricks/strings/group_by_lines.h:64]: :+1: c5ec7c52

File: bricks/strings/test.cc (hunk 3 above)
Comment (dkorolev) [reply bricks/strings/test.cc]: :+1: c5ec7c52

**Comment by dkorolev** [comment]
TY @mzhurovich, PTAL.

**Pull request merged**